kratos/linear_solvers/linear_solver.h

```diff
@@ -257,48 +254,115 @@ class LinearSolver
         mpReorderer = pNewReorderer;
     }
 
+    /**
+     * @brief This method allows to set the tolerance in the linear solver
+     * @param NewTolerance The new tolerance set
+     */
     virtual void SetTolerance(double NewTolerance)
     {
-        std::cout << "WARNING: Accessed base function Kratos::LinearSolver::SetTolerance(double). This does nothing !" << std::endl;
+        KRATOS_WARNING("LinearSolver") << "WARNING: Accessed base function Kratos::LinearSolver::SetTolerance(double). This does nothing !" << std::endl;
     }
-
+    
+    /**
+     * @brief This method allows to get the tolerance in the linear solver
+     * @return The tolerance
+     */
     virtual double GetTolerance()
     {
-        std::cout << "WARNING: Accessed base function Kratos::LinearSolver::GetTolerance(). No tolerance defined, returning 0 !" << std::endl ;
+        KRATOS_WARNING("LinearSolver") << "WARNING: Accessed base function Kratos::LinearSolver::GetTolerance(). No tolerance defined, returning 0 !" << std::endl ;
         return 0;
     }
 
-
     ///@}
     ///@name Inquiry
     ///@{
 
-    virtual bool IsConsistent(SparseMatrixType& rA, VectorType& rX, VectorType& rB)
+    /**
+     * @brief This method checks if the dimensions of the system of equations are consistent
+     * @param rA The LHS of the system of equations
+     * @param rX The vector containing the unkowns
+     * @param rB The RHS of the system of equations
+     * @return True if consistent, false otherwise
+     */
+    virtual bool IsConsistent(
+        SparseMatrixType& rA, 
+        VectorType& rX, 
+        VectorType& rB
+        )
     {
         const SizeType size = TSparseSpaceType::Size1(rA);
-
-        return ((size ==  TSparseSpaceType::Size2(rA)) &&
-                (size ==  TSparseSpaceType::Size(rX)) &&
-                (size ==  TSparseSpaceType::Size(rB)));
+        const SizeType size_a = TSparseSpaceType::Size2(rA);
+        const SizeType size_x = TSparseSpaceType::Size(rX);
+        const SizeType size_b = TSparseSpaceType::Size(rB);
+
+        KRATOS_WARNING_IF("LinearSolver", size !=  size_a) << "WARNING: Your LHS matrix is not square. Sizes: " << size << " vs " << size_a << std::endl ;
+        KRATOS_WARNING_IF("LinearSolver", size !=  size_x) << "WARNING: Your unkowns vector is not properly sized. Sizes: " << size << " vs " << size_x << std::endl ;
+        KRATOS_WARNING_IF("LinearSolver", size !=  size_b) << "WARNING: Your RHS vector is not properly sized. Sizes: " << size << " vs " << size_b << std::endl ;
+        
+        return ((size ==  size_a) &&
+                (size ==  size_x) &&
+                (size ==  size_b));
     }
 
-    virtual bool IsConsistent(SparseMatrixType& rA, DenseMatrixType& rX, DenseMatrixType& rB)
+    /**
+     * @brief This method checks if the dimensions of the system of equations are consistent (dense matrix version)
```

Comment:
here you say dense matrix version but the argument is a sparse matrix

Reply:
The rX and Rb are dense matrices instead of vectors